Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,125 @@
+        } catch {
+            NSLog("Could not migrate database to app group " + error.localizedDescription)
+            CrashLogging.logError(error)
+
+            removeFailedMigrationFilesIfNeeded()
+            return .failed
+        }
+    }
+
+    /// This method disables journaling on the core data database
+    /// Per this doc: https://developer.apple.com/library/archive/qa/qa1809/_index.html
+    /// Core data databases have journaling enabled by default, without disabling the journaling first it is possible some notes may get lost in migration
+    ///
+    private func disableJournaling() throws {
+        NSLog("Attempting to disable journaling on persistent store at \(storageSettings.legacyStorageURL)")
+        try loadPersistentStorage(at: storageSettings.legacyStorageURL, journaling: false)
+    }
+
+    private func loadPersistentStorage(at storagePath: URL, journaling: Bool) throws {
+        guard let mom = NSManagedObjectModel(contentsOf: storageSettings.modelURL) else {
+            throw NSError(domain: "SharedStorageMigrator", code: 100, userInfo: nil)
+        }
+        let psc = NSPersistentStoreCoordinator(managedObjectModel: mom)
+
+        let options = journaling ? nil : [NSSQLitePragmasOption: Constants.journalModeDisabled]
+
+        try psc.addPersistentStore(ofType: NSSQLiteStoreType,
+                                   configurationName: nil,
+                                   at: storagePath,
+                                   options: options)
+
+        // Remove the persistent store before exiting
+        // If removing fails, the migration can still continue so not throwing the errors
+        do {
+            for store in psc.persistentStores {
+                try psc.remove(store)
+            }
+        } catch {
+            NSLog("Could not remove temporary persistent Store " + error.localizedDescription)
+        }
+    }
+
+    private func migrateCoreDataFiles() throws {
+        try fileManager.copyItem(at: storageSettings.legacyStorageURL, to: storageSettings.sharedStorageURL)
+    }
+
+    private func attemptCreationOfCoreDataStack() throws {
+        NSLog("Confirming migrated database can be loaded at: \(storageSettings.sharedStorageURL)")
+        try loadPersistentStorage(at: storageSettings.sharedStorageURL, journaling: true)
+    }
+
+    private func removeFailedMigrationFilesIfNeeded() {
+        guard sharedStorageExists else {
+            return
+        }
+
+        do {
+            try fileManager.removeItem(at: storageSettings.sharedStorageURL)
+        } catch {
+            NSLog("Could not delete files from failed migration " + error.localizedDescription)
+        }
+    }
+}
+
+enum MigrationResult {
+    case success
+    case notNeeded
+    case failed
+}
+
+private struct Constants {
+    static let journalMode = "journal_mode"
+    static let journalSetting = "DELETE"
+    static let journalModeDisabled = [Constants.journalMode: Constants.journalSetting]
```

Comment:
You can prob. just use `"journal_mode": "DELETE"` (since the sub-constants aren't used anywhere else)